de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java

```diff
@@ -182,10 +186,18 @@ public Status run(Map<String, IProject> projectMapping,
      * @param monitor
      *      monitor to show progress to the user
      *
-     * @throws IOException, SarosCancellationException
+     * @throws SarosCancellationException
      */
-    protected abstract void setup(IProgressMonitor monitor)
-        throws IOException, SarosCancellationException;
+    protected void setup(IProgressMonitor monitor)
```

Comment:
There is no description what this method does. Well it setup something, but what ? Furthermore if I would inherit from this class but do not want to use some SMACK API I have to NOP this method.

Reply:
Ok, then probably `AbstractOutgoingProjectNegotiation` should change too, but I actually have no good idea about a different "short" name for this.

Removing the Abstract is fine and make it more consistent with the rest of the package.
But `FileTransferBasedIncomingProjectNegotiation` is even longer than the old name. Just shorten it to `FTBIncomingProjectNegotiation`, wouldn't follow the naming convention advice.

Using `SmackIncomingProjectNegotiation` is shorter, would take the used library into account, but is a bad idea because smack could get changed.
And more generally, I think about refactoring the usage of the Smack `FileTransferListener`, because as you mentioned already in `SendFileAction`, the usage in the Eclipse Part is misplaced to. This could maybe lead to removing smack from this abstract class again.

So if you really want to change the name of it, I'm not preferring it and not sure if its worth the effort.
But I don't want to negotiate more about it and if you like, just rename
`AbstractIncomingProjectNegotiation` -> `FileTransferBasedIncomingProjectNegotiation`
`AbstractOutgoingProjectNegotiation` -> `FileTransferBasedOutgoingProjectNegotiation`

What do you think?

Reply:
You could also remove just the Abstract part. However, let's do this in another patch.